templates/a_muon.cpp.in

```diff
@@ -33,15 +33,233 @@
+ *
+ * This template evaluates the contribution to the magnetic
+ * dipole moment of a BarZee diagram with fields given by
+ * \a Args.
+ */
+template<class PhotonEmitter, class ExchangeParticle>
+struct AMuonBarZeeVectorLoop {
+   static double value(const typename field_indices<Muon>::type& indices,
+                       const context_base& context,
+                       const softsusy::QedQcd& qedqcd);
+};
+
+/**
+    Loop function needed for the calculation of the BarZeeDiagram with
+    fermion loop. (arXiv:1502.04199 Equation 26)
+
+    @param respective squared mass ratio.
+*/
+
+double BarZeeLoopFPS(double m) {
+   if (m == 0) {
+      return 0;
+   }
+
+   if (m == 0.25) {
+      return log(2.0);
+   }
+
+   double r1, theta1, r2, theta2;
+   if (m <= 0.25) {
+      const double y = std::sqrt(1.0-4.0*m);
+      r1 = 1.0 - (1.0 - y)/(2*m);
+      r2 = 1.0 - (1.0 + y)/(2*m);
+      if (r1 > 0) {
+         theta1 = 0.0;
+      }
+      else {
+         r1 *= -1;
+         theta1 = M_PI;
+      }
+      if (r2 > 0) {
+         theta2 = 0.0;
+      }
+      else {
+         r2 *= -1;
+         theta2 = M_PI;
+      }
+      return m / y * (dilog(std::polar(r1, theta1)).real() - dilog(std::polar(r2, theta2)).real());
+   }
+   else {
+      const double y = std::sqrt(-1.0+4.0*m);
+      const double real = 1 - 1/(2*m);
+      r1 = 1.0;
+      r2 = 1.0;
+
+      theta1 = std::atan2(y/(2*m), real);
+      theta2 = std::atan2(-y/(2*m), real);
+
+      return m / y * (dilog(std::polar(r1, theta1)).imag() - dilog(std::polar(r2, theta2)).imag());
+   }
+}
+
+/**
+    Loop function needed for the calculation of the BarZeeDiagram with
+    fermion loop. (arXiv:1502.04199 Equation 25)
+
+    @param respective squared mass ratio.
+*/
+
+double BarZeeLoopFS(double m)
+{
+    if (m == 0)
+        return 0;
+
+    return (2*m - 1) * BarZeeLoopFPS(m) - m * (2 + log(m));
+}
+
+/**
+    Loop function needed for the calculation of the BarZeeDiagram with
+    fermion loop. (arXiv:1502.04199 Equation 27)
+
+    @param respective squared mass ratio.
+*/
+
+double BarZeeLoopS(const double m)
+{
+   return 1 + log(m)/2 - BarZeeLoopFPS(m);
+}
+
+/**
+    Loop function needed for the calculation of the BarZeeDiagram with
+    fermion loop. (arXiv:1502.04199 Equation 28)
+
+    @param respective squared mass ratio.
+*/
+
+double BarZeeLoopV(const double m) {
+   if (m == 0.25) {
+      return 4.75;
+   }
+
+   double j;
+
+   double r1, theta1, r2, theta2;
+
+   if (m < 0.25) {
+      const double y = std::sqrt(1.0-4.0*m);
+      r1 = 2 / (1-y);
+      theta1 = std::atan2(0, r1);
+      r1 = std::abs(r1);
+      r2 = 2 / (1+y);
+      theta2 = std::atan2(0, r2);
+      r2 = std::abs(r2);
+
+      j = 1/y * (std::log(std::abs(y-1) / (y+1)) * std::log(m) + dilog(std::polar(r1, theta1)).real() - dilog(std::polar(r2, theta2)).real());
+   }
+   else {
+      const double y = std::sqrt(-1.0+4.0*m);
+      r1 = std::sqrt(1/m);
+      r2 = std::sqrt(1/m);
```

Comment:
These two lines are identical, so we could simply write
~~~.cpp
r1 = r2 = std::sqrt(1/m);
~~~

Reply:
Done in b7c459e675f77c3d3c53830dedd387a012e39c6e.